src/utils.ts

```diff
@@ -51,16 +56,18 @@ export async function fetchData<T>(url: string, body?: unknown): Promise<T> {
   }
 
   const resp = await fetch(url, options)
-  const json = await resp.json()
+  let json
 
-  if (!resp.ok) {
-    let errTxt = ''
-    try {
-      const err = json as ErrorResponse
-      errTxt = `${err.code}: ${err.message}`
-    } catch (e) {
-      errTxt = resp.statusText
+  try {
+    json = await resp.json()
+  } catch {
+    if (resp.headers && resp.headers.get('content-length') === '0') {
+      throw new Error('Empty response. ' + resp.statusText)
```

Comment:
I think it should be like this:
```suggestion
    if (resp.headers && resp.headers.get('content-length') !== '0') {
      throw new Error('Response has invalid format: ' + resp.text())
```

Otherwise we would throw if a response is empty (which we want to avoid with this PR)

Reply:
Ah wait. After calling `resp.json()` we cannot call `resp.text()` anymore. So i would probably just throw like this:
`throw new Error('Response has invalid format: ')`

Reply:
Good catch. I've adjusted it as suggested.